chemprop/v2/data/utils.py

```diff
@@ -2,65 +2,253 @@
 from typing import Sequence
 
 import numpy as np
-from rdkit.Chem.Scaffolds import MurckoScaffold
 
 
 from chemprop.v2.data.datapoints import MoleculeDatapoint
+from chemprop.v2.data.datasets import MoleculeDataset
+
+from astartes import train_val_test_split, train_test_split
+from astartes.molecules import train_val_test_split_molecules, train_test_split_molecules
+
+import copy
+from logging import Logger
+import pickle
+from random import Random
+from typing import Tuple
+import os
+
+from chemprop.data import log_scaffold_stats
+from chemprop.args import TrainArgs
 
 
 def split_data(
     data: Sequence[MoleculeDatapoint],
     split: str = "random",
-    sizes: tuple[float, float, float] = (0.8, 0.1, 0.1),
-    k: int = 5,
-    fold: int = 0,
-):
+    sizes: Tuple[float, float, float] = (0.8, 0.1, 0.1),
+    key_molecule_index: int = 0,
+    seed: int = 0,
+    num_folds: int = 1,
+    args: TrainArgs = None,
```

Comment:
The 'TrainArgs' class is no more. We'll have to find another way to get the args needed into this function.

Reply:
41d28da addresses this. Unforunately train args was used for three of the splitting approaches (different ways of loading splits from files) so we will indeed need to add that back. I think we should only add back one way to do it tough. v1 had three approaches which did slightly different things and were all poorly documented.

Reply:
This will be added back in another PR, after more discussion with the team about the best way to do it.